src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -121,4 +127,49 @@ public abstract partial class DiagnosticSource
         public System.Diagnostics.Activity StartActivity(System.Diagnostics.Activity activity, object? args) { throw null; }
         public void StopActivity(System.Diagnostics.Activity activity, object? args) { }
     }
+    public enum ActivitySourceEventOperation
+    {
+        SourceCreated = 0,
+        ActivityStarted = 1,
+        ActivityStopped = 2
+    }
+    public enum ActivityKind
+    {
+        Internal = 1,
+        Server = 2,
+        Client = 3,
+        Producer = 4,
+        Consumer = 5,
+    }
+    public sealed class ActivitySourceEventArgs : System.EventArgs
+    {
+        private ActivitySourceEventArgs() { throw null; }
+        public System.Diagnostics.ActivitySourceEventOperation Operation { get; }
+    }
+    public readonly struct ActivityContext
+    {
+        public ActivityContext(System.Diagnostics.ActivityTraceId traceId, System.Diagnostics.ActivitySpanId spanId, System.Diagnostics.ActivityTraceFlags traceOptions, string? traceState = null) { throw null; }
+        public System.Diagnostics.ActivityTraceId TraceId { get; }
+        public System.Diagnostics.ActivitySpanId SpanId { get; }
+        public System.Diagnostics.ActivityTraceFlags TraceFlags { get; }
+        public string? TraceState { get; }
+    }
+    public readonly struct ActivityLink
+    {
+        public ActivityLink(System.Diagnostics.ActivityContext context) { throw null; }
+        public ActivityLink(System.Diagnostics.ActivityContext context, System.Collections.Generic.IDictionary<string, object>? attributes) { throw null; }
+        public System.Diagnostics.ActivityContext Context { get; }
+        public System.Collections.Generic.IDictionary<string, object>? Attributes { get; }
+    }
+    public sealed class ActivitySource : IDisposable
```

Comment:
we  discussed that for libs (Azure SDK) it's not a problem to create `ActivitySource` per operation, but it seems it could a problem/typical mistake for app developers.

E.g. imagine in the controller you want to track custom Activities. Developers would have to create static  `ActivitySource`  and I guess it will be common forget/not read docs and do this:

```csharp
using (var s = new ActivitySource("foo"))
using (var a = s.CreateActivity())
{
  // do stuff
}
```

Can we think about it from app dev convenience?

Reply:
does the concern here is ActivitySource is Disposable? if so, we can think in another way to dispose it. but wouldn't anyone writing a code as you showed will not work for them anyway and will be obvious there is a problem?

Reply:
> The only downside of that pattern is the app author is paying the cost to create the ActivitySource, publish it, and have telemetry agents decide if they want to subscribe to it on each Activity generated.

This is exactly my concern: the easiest and most convenient way to use this API requires 2 allocations and listener to start/stop subscription in e.g. per-request basis (imagine concurrency issues and locks in listener).

Assuming source is less granular that activity, it's possible to have 1) default source and you create one per app or web host 2) source per e.g. controller that you can register as singleton in your  container.

With current implementation it's not even possible to register source in DI container (if you want more than one) and users will be forced to have static fields to keep each source.

Reply:
A couple of clarification questions to get the whole picture: 
- Does the ask to make it easier to create a source, publish and listen to the custom activities without the need to hold the source in a static field (or store it somewhere)?
- if the answer is yes, does it require at any point later you can retrieve the exactly created source instance to do something else with it? or don't care about the source after establishing the event listeners?

Reply:
> Assuming source is less granular that activity, it's possible to have 1) default source and you create one per app or web host 2) source per e.g. controller that you can register as singleton in your container.

I think similar to @tarekgh I am trying to understand your goal : ) Is the goal of the default source to allow for simpler usage like this?

````
    using(Activity a = Activity.Start("activityName"))
    {
    }
````

The source per controller was more ambiguous what aspects you were finding appealing?

> With current implementation it's not even possible to register source in DI container

Is your concern that if we don't then it won't appeal to developers writing ASP.Net apps? Something else?

Reply:
with the design with so many allocations - is the assumption that the span name for ASP.NET incoming requests will be a constant value like `"HttpIn"`? In OpenTelemetry there is a notion of a name and for http it is typically the "route name". So we either need a notion of sub-name or `Activity.Name` is actually a "component name". And we are building a spec for "component"

Reply:
We are going to get rid of ActivitySource all together. I am going to share the new final proposal by tomorrow or so.

Reply:
Can we call ActivitySource `Tracer` in the new proposal?

Reply:
We are going to get rid of ActivitySource all together.